core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -252,6 +258,8 @@ private IndexMetaData(Index index, long version, State state, int numberOfShards
         this.indexCreatedVersion = indexCreatedVersion;
         this.indexUpgradedVersion = indexUpgradedVersion;
         this.minimumCompatibleLuceneVersion = minimumCompatibleLuceneVersion;
+        // by default we treat indices that start with '.' as hidden
+        this.hidden = INDEX_HIDDEN_SETTING.exists(settings) ? INDEX_HIDDEN_SETTING.get(settings) : index.getName().startsWith(".");
```

Comment:
question: will we see the setting hidden:true amongst the index settings, or will it be implicit?